eZ/Publish/Core/Search/Legacy/Content/FieldValueMapper/DateMapper.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/**
+ * This file is part of the eZ Publish Kernel package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\Search\Legacy\Content\FieldValueMapper;
+
+use eZ\Publish\Core\Search\Common\FieldValueMapper\DateMapper as CommonDateMapper;
+use eZ\Publish\SPI\Search\Field;
+use DateTime;
+use InvalidArgumentException;
+use Exception;
+
+/**
+ * Maps DateField document field values to something legacy search engine can index.
+ */
+class DateMapper extends CommonDateMapper
+{
+    /**
+     * Map field value to a proper legacy search engine representation.
+     *
+     * @param \eZ\Publish\SPI\Search\Field $field
+     *
+     * @return mixed
+     */
+    public function map(Field $field)
+    {
+        if (is_numeric($field->value)) {
+            $date = new DateTime("@{$field->value}");
```

Comment:
For the Legacy/SQL Search Engine there is no need to create `DateTime` object here, let's just `return $field->value;`. Fixed in c5746c1.